### PR TITLE
Remove Android Pre-AppCompat 

### PR DESCRIPTION
**Pull request by domagojmedo**
### Description of Change ###

This is a copy of https://github.com/dotnet/maui/pull/174 that removes Android Pre-AppCompat

### Issues Resolved ### 

- fixes https://github.com/dotnet/maui/issues/20 (couldn't find any related issue in XF repo)

### API Changes ###

Added:
    Moved methods and properties from Platform to AppCompat.Platform:
-  internal const string AppCompat.Platform.CloseContextActionsSignalName 
-  internal static readonly BindableProperty AppCompat.Platform.RendererProperty
-  public static void ClearRenderer(AView renderedView)
-  public static IVisualElementRenderer CreateRenderer(VisualElement element)
-  internal static IVisualElementRenderer CreateRenderer(VisualElement element, Context context)
-  internal static IVisualElementRenderer CreateRenderer(VisualElement element, FragmentManager fragmentManager, Context context)
-  public static IVisualElementRenderer CreateRendererWithContext(VisualElement element, Context context)
-  public static IVisualElementRenderer GetRenderer(VisualElement bindable)
-  public static void SetRenderer(VisualElement bindable, IVisualElementRenderer value)
-  internal ViewGroup GetViewGroup()
-  internal static readonly BindableProperty PageContextProperty 
-  internal static void SetPageContext(BindableObject bindable, Context context)
-  static Context GetPreviewerContext(Element element)
 
 Removed clases:
- Xamarin.Forms.ControlGallery.Android.FormsApplicationActivity
- Xamarin.Forms.Platform.Android.FormsApplicationActivity
- Xamarin.Forms.Platform.Android.Platform
- Xamarin.Forms.Platform.Android.ButtonRenderer
- Xamarin.Forms.Platform.Android.CarouselPageRenderer
- Xamarin.Forms.Platform.Android.FrameRenderer
- Xamarin.Forms.Platform.Android.MasterDetailRenderer
- Xamarin.Forms.Platform.Android.NavigationRenderer
- Xamarin.Forms.Platform.Android.PickerRenderer
- Xamarin.Forms.Platform.Android.NavigationRenderer
- Xamarin.Forms.Platform.Android.SwitchRenderer
- Xamarin.Forms.Platform.Android.TabbedRenderer
 
### Platforms Affected ### 

- Android

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.ControlGallery.Android/RegistrarValidationService.cs [hunk 1]
```diff
@@ -21,15 +21,15 @@ public bool Validate(VisualElement element, out string message)
 			if (element == null)
 				return true;
 
-			var renderer = Platform.Android.Platform.CreateRendererWithContext(element, _context);
+			//var renderer = Platform.Android.Platform.CreateRendererWithContext(element, _context);
```
Comment (domagojmedo) [inline Xamarin.Forms.ControlGallery.Android/RegistrarValidationService.cs:24]: Pre-AppCompat Platform was public, now it's internal, not sure how to resolve this

File: Xamarin.Forms.ControlGallery.Android/Tests/TestingPlatformService.cs [hunk 2]
```diff
@@ -11,9 +10,11 @@ class TestingPlatformService : ITestingPlatformService
 	{
 		public async Task CreateRenderer(VisualElement visualElement)
 		{
-			await Device.InvokeOnMainThreadAsync(() => 
-				Platform.Android.Platform.CreateRendererWithContext(visualElement,
-					DependencyService.Resolve<Context>()));
+			//await Device.InvokeOnMainThreadAsync(() =>
```
Comment (domagojmedo) [inline Xamarin.Forms.ControlGallery.Android/Tests/TestingPlatformService.cs:13]: Pre-AppCompat Platform was public, now it's internal, not sure how to resolve this

**Comment by PureWeen** [comment]
@domagojmedo can you rebase this to 5.0.0 ?

**Comment by domagojmedo** [comment]
@PureWeen TBH at work we always do merges instead of rebase, I tried rebasing but I think I did something wrong

**Comment by PureWeen** [comment]
@domagojmedo yea retargeting the branch is tricky

Do you want to just get it back targeted to main and then I can retarget the branch?

**Comment by domagojmedo** [comment]
@PureWeen I can close this MR and open new one with just my commits on 5.0.0 branch instead of main. Would that work?

**Comment by PureWeen** [comment]
> @PureWeen I can close this MR and open new one with just my commits on 5.0.0 branch instead of main. Would that work?
> That way you wouldn't have to do anything

That works!!

**Comment by domagojmedo** [comment]
@PureWeen done https://github.com/xamarin/Xamarin.Forms/pull/12103

**Pull request closed without merging**